WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/Create Shipping Label Form/ShippingLabelFormViewController.swift

```diff
@@ -281,7 +281,17 @@ private extension ShippingLabelFormViewController {
         } onSwitchChange: { (switchIsOn) in
             // TODO: Handle order completion
         } onButtonTouchUp: {
-            // TODO: Purchase Label action
+            self.displayPurchaseProgressView()
+            self.viewModel.purchaseLabel { [weak self] result in
+                switch result {
+                case .success:
+                    self?.displayPrintShippingLabelVC()
+                case .failure:
+                    // TODO: Implement and display error screen for purchase failures
```

Comment:
Should we add this task as part of the list? https://github.com/woocommerce/woocommerce-ios/issues/4089

Reply:
Yep, it's there! This is the `UI for failed purchase` task in that list.